src/Compilers/Core/Portable/Operations/OperationInterfaces.xml

```diff
@@ -3057,4 +3057,111 @@
+          or an <see cref="IInterpolatedStringAdditionOperation"/> operation.
+        </summary>
+      </Comments>
+    </Property>
+  </Node>
+  <Node Name="IInterpolatedStringAdditionOperation" Base="IOperation" ChildrenOrder="Left,Right">
+    <Comments>
+      <summary>
+        Represents an addition of multiple interpolated string literals being converted to an interpolated string handler type.
+      </summary>
+    </Comments>
+    <Property Name="Left" Type="IOperation">
+      <Comments>
+        <summary>
+          The interpolated string expression or addition operation on the left side of the operator. This is either an <see cref="IInterpolatedStringOperation"/>
+          or an <see cref="IInterpolatedStringAdditionOperation"/> operation.
+        </summary>
+      </Comments>
+    </Property>
+    <Property Name="Right" Type="IOperation">
+      <Comments>
+        <summary>
+          The interpolated string expression or addition operation on the right side of the operator. This is either an <see cref="IInterpolatedStringOperation"/>
+          or an <see cref="IInterpolatedStringAdditionOperation"/> operation.
+        </summary>
+      </Comments>
+    </Property>
+  </Node>
+  <Node Name="IInterpolatedStringAppendOperation" Base="IInterpolatedStringContentOperation">
+    <OperationKind>
+      <Entry Name="InterpolatedStringAppendLiteral" Value="0x74" ExtraDescription=""/>
+      <Entry Name="InterpolatedStringAppendFormatted" Value="0x75" ExtraDescription="This append is of an interpolation component"/>
+    </OperationKind>
+    <Comments>
+      <summary>Represents a call to either AppendLiteral or AppendFormatted as part of an interpolated string handler conversion.</summary>
+    </Comments>
+    <Property Name="AppendCall" Type="IOperation?">
+      <Comments>
+        <summary>
+          If this interpolated string is subject to an interpolated string handler conversion, the construction of the interpolated string handler instance.
+          This can be an IInvocationOperation or IDynamicInvocationOperation for valid code, and IInvalidOperation for invalid code.
+        </summary>
+      </Comments>
+    </Property>
+  </Node>
+  <Node Name="IInterpolatedStringHandlerArgumentPlaceholderOperation" Base="IOperation">
+    <Comments>
+      <summary>Represents an argument from the method call, indexer access, or constructor invocation that is creating the containing <see cref="IInterpolatedStringHandlerCreationOperation"/></summary>
+    </Comments>
+    <Property Name="ArgumentIndex" Type="int">
+      <Comments>
+        <summary>
+          The index of the argument of the method call containing the interpolated string handler conversion this placeholder is referencing. -1 if
+          <see cref="IsContainingMethodReceiver"/> or if the argument was not resolved.
+        </summary>
+      </Comments>
+    </Property>
+    <Property Name="IsContainingMethodReceiver" Type="bool">
```

Comment:
><Property Name="IsContainingMethodReceiver" Type="bool"> [](http://example.com/codeflow?start=4&length=56)

Can we use special ArgumentIndex value instead of having a dedicated property?

Reply:
We could, though I wanted to avoid having a special constant for `Bad` vs `ContainingMethodReceiver` here for ergonomics.